src/synthesizer/dust/attenuation.py

```diff
@@ -39,26 +39,35 @@ def N09_tau(lam, slope, cent_lam, ampl, gamma):
     """
 
     # Wavelength in microns
-    tmp_lam = np.arange(0.12, 2.2, 0.001)
+    lam_micron = lam.to(um).value
     lam_v = 0.55
-    k_lam = np.zeros_like(tmp_lam)
-
-    ok = (tmp_lam < 0.63)
-    k_lam[ok] = -2.156 + (1.509 / tmp_lam[ok]) \
-        - (0.198 / tmp_lam[ok] ** 2) \
-        + (0.011 / tmp_lam[ok] ** 3)
-    k_lam[~ok] = -1.857 + (1.040 / tmp_lam[~ok])
+    k_lam = np.zeros_like(lam_micron)
+
+    ok1 = (lam_micron >= 0.12) * (lam_micron < 0.63)
```

Comment:
could you provide some brief comments here as to what the `ok1`, `ok2` and `ok3` masks represent?

Reply:
Added description